# Source: https://gitlab.com/gitlab-org/gitlab/-/blob/v19.1.2/Gemfile
# CVE-2025-67202 requires sidekiq-cron 2.4.0: https://nvd.nist.gov/vuln/detail/CVE-2025-67202
# CVE-2026-44476 requires doorkeeper-openid_connect 1.10.0: https://nvd.nist.gov/vuln/detail/CVE-2026-44476
# sidekiq-cron 2.4 compatibility: https://github.com/sidekiq-cron/sidekiq-cron/compare/v2.3.1...v2.4.0
diff --git a/Gemfile b/Gemfile
--- a/Gemfile
+++ b/Gemfile
@@ -88,7 +88,7 @@ gem 'devise-pbkdf2-encryptable', '~> 0.0.0', path: 'vendor/gems/devise-pbkdf2-en
   feature_category: :system_access
 gem 'bcrypt', '~> 3.1', '>= 3.1.14', feature_category: :system_access
 gem 'doorkeeper', '~> 5.8', '>= 5.8.1', feature_category: :system_access
-gem 'doorkeeper-openid_connect', '~> 1.9.0', feature_category: :system_access
+gem 'doorkeeper-openid_connect', '~> 1.10.0', feature_category: :system_access
 gem 'doorkeeper-device_authorization_grant', '~> 1.0.0', feature_category: :system_access
 gem 'rexml', '~> 3.4.0', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
 gem 'ruby-saml', '~> 1.18', feature_category: :system_access
@@ -304,7 +304,7 @@ gem 'sidekiq', path: 'vendor/gems/sidekiq', require: 'sidekiq', feature_category
 
 # Background jobs
 gem 'sidekiq', path: 'vendor/gems/sidekiq', require: 'sidekiq', feature_category: :scalability
-gem 'sidekiq-cron', '~> 2.3.0', feature_category: :scalability
+gem 'sidekiq-cron', '~> 2.4.0', feature_category: :scalability
 gem 'gitlab-sidekiq-fetcher',
   path: 'vendor/gems/sidekiq-reliable-fetch',
   require: 'sidekiq-reliable-fetch',
diff --git a/lib/gitlab/patch/sidekiq_cron_poller.rb b/lib/gitlab/patch/sidekiq_cron_poller.rb
--- a/lib/gitlab/patch/sidekiq_cron_poller.rb
+++ b/lib/gitlab/patch/sidekiq_cron_poller.rb
@@ -13 +13 @@
-if Gem::Version.new(Sidekiq::Cron::VERSION) != Gem::Version.new('2.3.1')
+if Gem::Version.new(Sidekiq::Cron::VERSION) != Gem::Version.new('2.4.0')
